xml-conduit fails to compile with ghc-8.rc2 (type errors)

note: issue #71 (which links to  https://ghc.haskell.org/trac/ghc/ticket/11276) has a similar title, but seems to different error (#71: type checker hangs, this: type checking results in errors)

xml-conduit commit 498b81b1166b72585ff1819521e5534a86cd3e86 ,
ghc --version 8.0.0.20160204 (rc2) 

```
cabal install:
(conflict: xml-conduit => transformers>=0.2 && <0.5)
...
```
```
cabal install --allow-newer:

[4 of 7] Compiling Text.XML.Stream.Parse ( Text/XML/Stream/Parse.hs, dist/dist-sandbox-c639d5f8/build/Text/XML/Stream/Parse.o )

Text/XML/Stream/Parse.hs:204:28: error:
    • Couldn't match type ‘a1’ with ‘a0 -> [(TName, [Content])]’
      ‘a1’ is a rigid type variable bound by
        a type expected by the context:
          forall a1. a1 -> a1
        at Text/XML/Stream/Parse.hs:204:28
      Expected type: forall a. a -> a
        Actual type: (a0 -> [(TName, [Content])])
                     -> a0 -> [(TName, [Content])]
```

 
There are in fact a lot more errors shown, but I don't believe them, because of https://ghc.haskell.org/trac/ghc/ticket/11541 (fixed in HEAD,which I don't have, and present in rc2)

--- a/xml-conduit/Text/XML/Stream/Parse.hs
+++ b/xml-conduit/Text/XML/Stream/Parse.hs
@@ -1,7 +1,6 @@
 {-# LANGUAGE BangPatterns       #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleContexts   #-}
-{-# LANGUAGE ImpredicativeTypes #-}
 {-# LANGUAGE OverloadedStrings  #-}
 {-# LANGUAGE PatternGuards      #-}
 {-# LANGUAGE RankNTypes         #-}

